projects/cds/src/cms-components/profile-tag/profile-tag.component.spec.ts

```diff
@@ -0,0 +1,43 @@
+import { async, ComponentFixture, TestBed } from '@angular/core/testing';
+import { Observable, of } from 'rxjs';
+import { ProfileTagComponent } from './profile-tag.component';
+import { ProfileTagInjector } from './profile-tag.injector';
+
+class MockProfileTagInjector {
+  injectScript(): Observable<boolean[]> {
+    return of([true, true]);
+  }
+}
+
+describe('ProfileTagComponent', () => {
+  let component: ProfileTagComponent;
+  let fixture: ComponentFixture<ProfileTagComponent>;
+
+  beforeEach(async(() => {
+    TestBed.configureTestingModule({
+      declarations: [ProfileTagComponent],
+      providers: [
+        {
+          provide: ProfileTagInjector,
+          useClass: MockProfileTagInjector,
+        },
+      ],
+    }).compileComponents();
+  }));
+
+  beforeEach(() => {
+    fixture = TestBed.createComponent(ProfileTagComponent);
+    component = fixture.componentInstance;
+    fixture.detectChanges();
+  });
+
+  it('should create the component', () => {
+    expect(component).toBeTruthy();
+  });
+
+  it('should be loaded', () => {
+    let result: boolean;
+    component.profileTagEnabled$.subscribe(data => (result = Boolean(data)));
```

Comment:
Even though it's a test, it's a good practice to unsubscribe. Again, at the end it doesn't really matter.

A better question is: you are basically testing here do you have an array (Boolean([ ... ]). Is that how you want this to work? Or is the idea to enable tracking if either of the `profileTagEnabled$`'s combined streams is emits true. Something along the lines: `([changed, loaded]) => changed || loaded`

Reply:
good catch. I didn't update this after combining the streams  😞

Reply:
All good :)

Reply:
Actually, now that I look at it, I think this is ok. This is my merged stream, and it puts out either an array of booleans, or an NGRouterEvent. I actually don't care about the values outputted, as I don't do anything with them and just have side-effects within my stream (I.E. tap) . So I guess just checking that an array exists would be ok. What do you think?

Reply:
Ok, in that case the client (in this context the code that's using the `projects/cds/src/cms-components/profile-tag/profile-tag.injector.ts`) should not be concerned about the data structures being returned nor its types, right?
All that the client code cares about is when the injector emits i.e. a boolean `true` value, in which case it should do something (i.e. the component should render the html). Correct?

If that's the case, then we should consider simplifying the `profile-tag.injector.ts` to expose only one stream (i.e. injectScript() should be public, while `isProfileTagLoaded$` and `startTracking$` should be private). The returning type should simply be `boolean`.

We could probably simplify it even more, by removing `startTracking$` and using `combineLatest([pageLoaded$, consentChanged$])` inside of `injectScript()`.

What do you think?

Reply:
For some reason I just now got the email about this comment. Good point about making the stream only return boolean. Also with making the rest private.... not sure how I overlooked that :grin: